### PR TITLE
feat(scan): allow configuring for a specific precinct

### LINKED ISSUE #513: precinct-scanner: configure for a particular precinct
lock it down to a precinct

### PR DESCRIPTION
**Pull request by eventualbuddha**
At the moment this only prevents counting sheets that don't match the current precinct. It does not zero or close polls or anything like that yet.

https://user-images.githubusercontent.com/1938/118735420-435d7180-b7f5-11eb-96b3-f46908011e7a.mov

Closes #513

### CODE REVIEW

File: apps/precinct-scanner/src/AppRoot.tsx [hunk 1]
```diff
@@ -756,9 +763,10 @@ const AppRoot: React.FC<Props> = ({ hardware, card }) => {
     }
   }, [startUsbStatusPolling, startCardShortValueReadPolling])
 
-  const updatePrecinctId = (precinctId: string) => {
+  const updatePrecinctId = useCallback(async (precinctId: string) => {
     dispatchAppState({ type: 'updatePrecinctId', precinctId })
-  }
+    await config.setCurrentPrecinctId(precinctId)
+  }, [])
```
Comment (carolinemodic) [inline apps/precinct-scanner/src/AppRoot.tsx:769]: should dispatchAppState be a dependency here, I see you added it in other places?

File: apps/precinct-scanner/src/screens/InsertBallotScreen.tsx [hunk 2]
```diff
@@ -9,14 +9,19 @@ import * as format from '../utils/format'
...
 const InsertBallotScreen: React.FC<Props> = ({
   scannedBallotCount,
   electionDefinition,
+  currentPrecinctId,
 }) => {
   return (
-    <CenteredScreen electionDefinition={electionDefinition}>
+    <CenteredScreen
+      electionDefinition={electionDefinition}
+      currentPrecinctId={currentPrecinctId}
```
Comment (carolinemodic) [inline apps/precinct-scanner/src/screens/InsertBallotScreen.tsx:23]: As we talked about before FYI I might refactor this to put these two things and machineConfig in a context... in general I was trying out uses parameters more then a context in this app but I think when things are being passed as params to a layout component like this its probably better in a context.

**Review by carolinemodic — APPROVED** [review]

File: apps/precinct-scanner/src/AppRoot.tsx (hunk 1 above)
Comment (eventualbuddha) [inline apps/precinct-scanner/src/AppRoot.tsx:769]:
```suggestion
  }, [dispatchAppState])
```

File: apps/precinct-scanner/src/screens/InsertBallotScreen.tsx (hunk 2 above)
Comment (eventualbuddha) [reply apps/precinct-scanner/src/screens/InsertBallotScreen.tsx:23]: Agreed :+1:

**Pull request merged**